### PR TITLE
Pivotal ID # 185982595: Submission Request Completion Percentage

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
https://www.pivotaltracker.com/story/show/185982595

- Implement a formula to calculate the completion percentage based on the current stage and file index
- Return this calculation as part of the submissions list endpoint
- Add tests
- Remove unused code

### CODE REVIEW

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceQueryService.kt [hunk 1]
```diff
@@ -73,14 +78,22 @@ internal class SubmissionMongoPersistenceQueryService(
...
 
         return requests
-            .map { serializationService.deserialize(it.submission.toString()) }
-            .map { it.asBasicSubmission(PROCESSING) }
+            .map { serializationService.deserialize(it.submission.toString()) to it.completionPercentage() }
+            .map { it.first.asBasicSubmission(PROCESSING, it.second) }
             .plus(findSubmissions(submissionFilter))
     }
 
+    private fun DocSubmissionRequest.completionPercentage(): Double {
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceQueryService.kt:86]: The reason why I didn't put this logic in the `BasicSubmission` class is that we use this object to map both, submissions and requests hence, we wouldn't have the required arguments for already existing submissions or even submissions that were submitted before the stages mechanism was established.

I could set default values for these cases but I just wanted to clarify that.

[the remark this answers is not in the document]
Comment (Juan-EBI) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceQueryService.kt:86]: I think for submission already submission
```
stage: The final one
totalFiles: any number maybe one?
currentFile: same any number that total Files. 
```

**Review by Juan-EBI — APPROVED** [review]
LGTM

**Comment by jhoanmanuelms** [comment]
Closed until it's required in the ST

**Pull request closed without merging**